tests/meshcop/commissioner.cpp

```diff
@@ -838,7 +832,7 @@ int CommissionerServe(Context &aContext)
 exit:
     otbrLog(OTBR_LOG_INFO, "CommissionerServe: result=%d", ret);
 
-    return ret;
+    return aContext.mState == kStateDone ? 0 : -1;
```

Comment:
why not return `ret` directly?

Reply:
Returning ret directly will need additional flags for the double loop

Reply:
A little confused about the function of the variable `ret` in your interpretation.
How about initializing `ret` as `-1 `and exit without reassign `ret` with `errno` when fails to create the socket in Line 746?